classes/item.rb

```diff
@@ -9,6 +9,11 @@ def initialize(publish_date)
     @archived = false
   end
 
+  def add_label(label)
+    label.add_items << self unless label.items.include? self
+    @label = label
+  end
```

Comment:
I like the way you dealt with this 
but this could be so much easier please look at my next comment